Browser_IDE/executionEnvironment_Internal.js

```diff
@@ -215,32 +215,39 @@ ReferenceError: test is not defined
 // Currently those are the only two forms supported, but this should account for the majority well enough.
 // It also doesn't parse the url style ones - it only needs to work for the local user's code, so good enough.
 
+
 function parseErrorStack(err){
     const stackParse = /(?:@|\()((?:[^;:`]|[:;`](?=.*(?:\/|\.)))*)`?;?:([0-9]*)/g;
     let stack = [...err.stack.matchAll(stackParse)];
 
     let stackIndex = 0;
 
-    //Should we limit this to only SplashKitArgumentError? (i.e if (err instanceof SplashKitArgumentError))
-
     // Unwind stack until we find user code:
     while(stackIndex < stack.length && !stack[stackIndex][1].startsWith(userCodeBlockIdentifier))
         stackIndex += 1;
 
-    if (stackIndex >= stack.length)
-        stackIndex = 0;
-
-    let lineNumber = stack[stackIndex][2];
+    // Include all the messages relevant to the user
+    let userStackEnd = stackIndex;
+    while(userStackEnd < stack.length && stack[userStackEnd][1].startsWith(userCodeBlockIdentifier))
+        userStackEnd += 1;
 
-    let file = stack[stackIndex][1];
+    stack = stack.slice(stackIndex, userStackEnd);
 
-    if (file.startsWith(userCodeBlockIdentifier))
-        lineNumber -= userCodeStartLineOffset;
+    stack.forEach(s => {
+        s[1] = s[1].slice(userCodeBlockIdentifier.length); // Slice off the userCodeBlockIdentifier
+        s[2] -= userCodeStartLineOffset; // Subtract userCodeStartLineOffset from line numbers
+    });
 
-    return {lineNumber, file};
+    
+    stack = stack.map(s => s[1] + ': line ' + s[2]).join('\n'); // Adjust the format here
+    return {lineNumber: stack[0][2], file: stack[0][1], stack};
```

Comment:
Few notes:
1. `stack = stack.map(s => s[1] + ': line ' + s[2]).join('\n');`  - the formatting is good! But this line makes the next line's `stack[0][2]` and `stack[0][1]` not work. This is why there was no code file/line number being shown in the error messages (no need to pass things like that off as design decisions, it's totally fine to say there's a bug still and you need help fixing it 😃). With this fixed the errors look a bit more complete, and it also brings back error highlighting in the code editors too.
2. the `file: ` part needs to return the file with the `userCodeBlockIndentifier` still attached or later functions have trouble.

Here's my suggested fix:
```suggestion
    let formattedStack = stack.map(s => s[1] + ': line ' + s[2]).join('\n'); // Adjust the format here
    return {lineNumber: stack[0][2], file: userCodeBlockIdentifier + stack[0][1], stack: formattedStack};
```